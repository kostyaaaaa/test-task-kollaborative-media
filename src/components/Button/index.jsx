import { func, node, string } from 'prop-types';
import React from 'react';

const Button = ({ children, onClick, className }) => {
  return (
    <button onClick={onClick} className={className}>
      {children}
    </button>
  );
};

Button.defaultProps = {
  className: '',
};

Button.propTypes = {
  children: node.isRequired,
  onClick: func.isRequired,
  className: string,
};

export default Button;
