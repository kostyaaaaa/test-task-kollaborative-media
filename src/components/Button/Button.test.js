import renderer from 'react-test-renderer';
import Button from './index';

const mockedOnClick = jest.fn();

const component = renderer.create(
  <Button onClick={mockedOnClick} className="text-xl">
    Click me
  </Button>,
);

it('should match snapshot', () => {
  expect(component).toMatchSnapshot();
});

it('should apply provided props', () => {
  expect(component.toTree().rendered.props.className).toEqual('text-xl');
  expect(component.toTree().rendered.props.children).toEqual('Click me');
});

it('should call provided onClick props', () => {
  expect(mockedOnClick).toHaveBeenCalledTimes(0);
  // Simulate a click event
  component.root.props.onClick();
  expect(mockedOnClick).toHaveBeenCalledTimes(1);
});
