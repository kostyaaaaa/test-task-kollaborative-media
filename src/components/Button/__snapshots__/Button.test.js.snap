// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<button
  className="text-xl"
  onClick={[MockFunction]}
>
  Click me
</button>
`;
